quinn-proto/src/shared.rs

```diff
@@ -248,6 +244,26 @@ impl EndpointConfig {
         }
         Ok(())
     }
+
+    /// Set the local cid length.
+    ///
+    /// This is the length of connection IDs for the endpoint.
+    /// This must be either 0 or between 4 and 18 inclusive. The length of the local connection IDs
+    /// constrains the amount of simultaneous connections the endpoint can maintain. The API user is
+    /// responsible for making sure that the pool is large enough to cover the intended usage.
+    pub fn with_local_cid_len(&mut self, local_cid_len: usize) -> &mut Self {
```

Comment:
> I think we should omit the with_ prefix here,

If we do it here, we should do it on all methods as well.

>  It's also confusing when combined with mutating setters, since IMO "with" does not connote in-place modification.

Setters are mutable anyway, we won't have immutable setters.  Therefore we just have to keep one naming convention and use for all setters.

Reply:
> If we do it here, we should do it on all methods as well.

Yes, by "here" I meant "in this PR." Apologies for the unclear phrasing.

> Setters are mutable anyway, we won't have immutable setters.

I expect `with_`-prefixed methods to take `self` by value and produce a new value as a result, rather than updating a pre-existing value in-place. This is one of two common patterns for builders.